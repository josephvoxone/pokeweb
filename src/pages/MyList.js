import React from 'react';
import CardOwned from '../components/CardOwned';
// import Loading from '../components/Loading';
import styled from '@emotion/styled'
import { DiscardPokemon, getPokemon } from '../service/providers/Pokemon';
import { useHistory } from 'react-router-dom';

const MyList = () => {

    const history = useHistory()
    const mypokemons = getPokemon

    const Empty = styled.div`
        min-height: 400px;
        display: grid;
        justify-content: center;
        align-content: center;
    `;
    const Title = styled.h1`
        text-transform: capitalize;
        margin-top: 0;
        margin-bottom: 5px;
        color: white;
    `;
    const Subtitle = styled.h4`
        margin-top: 0;
        margin-bottom: 5px;
        color: white;
        font-weight: 400;
    `;
    const List = styled.div`
        justify-content: center;
        align-items: center;
        flex-wrap: wrap;
        max-width: 100%;
        min-height: 400px;
        display: flex;
        width: 1000px;
        margin: 0 auto;
    `;

    const OnDelete = (nickname) => {
        DiscardPokemon(nickname)
        history.go(0)
        console.log(getPokemon)
    }

    if (!mypokemons.length) {
        return (
            <Empty key='1'>
                <Title>Oops!</Title>
                <Subtitle>seems you doesn't have any pokemon.</Subtitle>
            </Empty>
        )
    };
    if (mypokemons) {
        const datapokemon = mypokemons.map(({ id, nickname, pokemon }, index) => {
            return <CardOwned
                key={nickname}
                index={index}
                id={id}
                pokemon={pokemon}
                nickname={nickname}
                onClick={OnDelete}
            />
        })
        console.log(datapokemon)

        return (
            <>
                <List>
                    {datapokemon}
                </List>
            </>
        )
    }

}

export default MyList;