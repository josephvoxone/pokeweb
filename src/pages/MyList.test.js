import React from "react";
import { render, unmountComponentAtNode } from "react-dom";
import { act } from "react-dom/test-utils";
import MyList from './MyList';

let container = null;
beforeEach(() => {
    // setup a DOM element as a render target
    container = document.createElement("div");
    document.body.appendChild(container);
});

afterEach(() => {
    // cleanup on exiting
    unmountComponentAtNode(container);
    container.remove();
    container = null;
});

it("Test : My List Page", () => {
    {
        act(() => {
            render(
                <MyList />, container)
        })
        expect(container).toMatchSnapshot()
    }
});
