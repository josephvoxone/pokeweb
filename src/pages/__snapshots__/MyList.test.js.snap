// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test : My List Page 1`] = `
<div>
  <div
    class="css-17y4c5w"
  >
    <h1
      class="css-1wp9jp3"
    >
      Oops!
    </h1>
    <h4
      class="css-3606z2"
    >
      seems you doesn't have any pokemon.
    </h4>
  </div>
</div>
`;
